### PR TITLE
Add e2e test for connectivity adapter

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- Add new e2e test for connectivity adapter from compass.

**Related issue(s)**

See https://github.com/kyma-incubator/compass/issues/678

### CODE REVIEW

File: tests/end-to-end/external-solution-integration/README.md [hunk 1]
```diff
@@ -42,6 +42,21 @@ The test performs the following actions:
 11. Sends an event to the Application Gateway. 
 12. Verifies if the call from the lambda reached the test service.
 
+## run test locally
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/README.md]:
```suggestion
### Run test locally
```

File: tests/end-to-end/external-solution-integration/README.md [hunk 2]
```diff
@@ -42,6 +42,21 @@ The test performs the following actions:
...
+
+    dexSecretNamespace: "kyma-system"
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/README.md]: Why those lines are here and what they represent?

File: tests/end-to-end/external-solution-integration/cmd/runner/main.go [hunk 3]
```diff
@@ -38,6 +40,7 @@ func main() {
 		for name := range scenarios {
 			log.Infof(" - %s", name)
 		}
+		os.Exit(0)
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/cmd/runner/main.go]: I think it should be a fail

File: tests/end-to-end/external-solution-integration/internal/scenario/connectivity_adapter/connectivity_adapter_e2e.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		step.Parallel(
+			testsuite.NewCreateNamespace(s.testID, clients.CoreClientset.CoreV1().Namespaces()),
+			testsuite.NewRegisterLegacyApplicationInCompass(s.testID, compassClients.DirectorClient, state),
+		),
+		step.Parallel(
+			testsuite.NewCreateMapping(s.testID, clients.AppBrokerClientset.ApplicationconnectorV1alpha1().ApplicationMappings(s.testID)),
+			testsuite.NewStartTestServer(testService),
+			testsuite.NewConnectApplicationUsingCompassLegacy(compassClients.ConnectorClient, appConnector, compassClients.DirectorClient, state),
+		),
+		testsuite.NewRegisterLegacyServiceInCompass(s.testID, testService.GetInClusterTestServiceURL(), compassClients.DirectorClient, testService, state),
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/internal/scenario/connectivity_adapter/connectivity_adapter_e2e.go]: Consider changing the name, you are registering the application using legacy API, not registering a legacy application.

File: tests/end-to-end/external-solution-integration/pkg/testsuite/create_service_binding_usage.go [hunk 5]
```diff
@@ -55,7 +55,7 @@ func (s *CreateLambdaServiceBindingUsage) Run() error {
 	serviceBindingUsage := &serviceBindingUsageApi.ServiceBindingUsage{
 		TypeMeta: metav1.TypeMeta{Kind: "ServiceBindingUsage", APIVersion: serviceBindingUsageApi.SchemeGroupVersion.String()},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:   s.name,
+			Name:   s.name, // Tutaj wykorzsytaj event secvice class name
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/pkg/testsuite/create_service_binding_usage.go]: Please remove the comment ;)

File: tests/end-to-end/external-solution-integration/pkg/testsuite/register_service_legacy_compass.go [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/internal/example_schema"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/step"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/testkit"
+	"github.com/pkg/errors"
+)
+
+// RegisterApplicationInCompass is a step which registers new Application with API and Event in Compass
+type RegisterLegacyServiceInCompass struct {
+	name string
+	//applications acClient.ApplicationInterface
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/pkg/testsuite/register_service_legacy_compass.go]: Remove unused code

File: tests/end-to-end/external-solution-integration/pkg/testsuite/register_legacy_application_in_compass.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+	}
+	s.state.SetCompassAppID(app.ID)
+
+	return nil
+}
+
+// Cleanup removes all resources that may possibly created by the step
+func (s *RegisterLegacyApplicationInCompass) Cleanup() error {
+	if s.state.GetCompassAppID() != "" {
+		_, err := s.director.UnregisterApplication(s.state.GetCompassAppID())
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/pkg/testsuite/register_legacy_application_in_compass.go]: Maybe this should be done with a retry manner?

File: tests/end-to-end/external-solution-integration/pkg/testsuite/create_service_instance.go [hunk 8]
```diff
@@ -104,7 +94,7 @@ func (s *CreateServiceInstance) isServiceInstanceCreated() error {
 
 // Cleanup removes all resources that may possibly created by the step
 func (s *CreateServiceInstance) Cleanup() error {
-	err := s.serviceInstances.Delete(s.name, &v1.DeleteOptions{})
+	err := s.serviceInstances.Delete(s.instanceName, &v1.DeleteOptions{})
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/pkg/testsuite/create_service_instance.go]: Consider implementing it with a retry approach

File: tests/end-to-end/external-solution-integration/pkg/testsuite/connect_application_using_compass.go [hunk 9]
```diff
@@ -35,11 +35,13 @@ func (s ConnectApplicationUsingCompass) Name() string {
 
 // Run executes the step
 func (s ConnectApplicationUsingCompass) Run() error {
+	// Use this
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/pkg/testsuite/connect_application_using_compass.go]: Please remove comments

File: tests/end-to-end/external-solution-integration/pkg/testkit/testservice.go [hunk 10]
```diff
@@ -65,6 +65,7 @@ func (ts *TestService) CreateTestService() error {
 func (ts *TestService) checkValue() (int, error) {
 
 	url := ts.GetTestServiceURL()
+	fmt.Printf("counter service URL %s\n", url)
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/pkg/testkit/testservice.go]: leftover?

File: tests/end-to-end/external-solution-integration/pkg/testkit/registry.go [hunk 11]
```diff
@@ -35,6 +35,7 @@ func (rc *RegistryClient) RegisterService(service *ServiceDetails) (string, erro
 
 	request.Header.Add("Content-Type", "application/json")
 
+	fmt.Printf("Service request: %+v\n", request)
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/pkg/testkit/registry.go]: leftover?

File: tests/end-to-end/external-solution-integration/pkg/testkit/legacy_registry_client.go [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+		return "", errors.Wrap(err, "error response")
+	}
+
+	registerServiceResponse := &RegisterServiceResponse{}
+
+	err = json.NewDecoder(response.Body).Decode(registerServiceResponse)
+	if err != nil {
+		return "", errors.Wrap(err, "while decoding response body")
+	}
+	fmt.Printf("service class from compass id: %s", registerServiceResponse.ID)
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/pkg/testkit/legacy_registry_client.go]: leftover?

File: tests/end-to-end/external-solution-integration/pkg/testkit/legacy_registry_client.go [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+	}
+}
+
+func (lrc *LegacyRegistryClient) RegisterService(url string, service *ServiceDetails) (string, error) {
+	body, err := json.Marshal(service)
+	if err != nil {
+		return "", errors.Wrap(err, "while marshalling service payload")
+	}
+
+	fmt.Printf("payload:%s\n", string(body))
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/pkg/testkit/legacy_registry_client.go]: leftover?

File: tests/end-to-end/external-solution-integration/pkg/testkit/connector.go [hunk 14]
```diff
@@ -112,6 +113,7 @@ func (cc *ConnectorClient) GetInfo(url string) (*InfoResponse, error) {
 		return nil, err
 	}
 
+	fmt.Printf("%+v\n", infoResponse.Api.MetadataUrl)
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/pkg/testkit/connector.go]: leftover?

File: tests/end-to-end/external-solution-integration/README.md [hunk 15]
```diff
@@ -42,7 +42,28 @@ The test performs the following actions:
...
+Connectivity Adapter is a component which translates Application Registry and Application Connector REST API to Compass GraphQL API. 
+
+### Steps
+
+The test performs the following actions:
+1. Registers an empty Application in Compass.
+2. Creates the ApplicationMapping CR for the created Application in the `connectivity-adapter-e2e` Namespace.
+3. Deploys a lambda function in the `connectivity-adapter-e2e` Namespace.
+4. Starts a test service in the `connectivity-adapter-e2e` Namespace. The lambda function calls it when it receives an event.
+6. Connects the Application through the Connectivity Adapter with the client certificates.
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/README.md]: Hmmm, should be 5 and then 6 ;)

File: tests/end-to-end/external-solution-integration/internal/scenario/connectivity_adapter_e2e/connectivity_adapter_e2e.go [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	clients := testkit.InitKymaClients(config, s.testID)
+	compassClients := testkit.InitCompassClients(clients, state, s.domain, s.skipSSLVerify)
+
+	connectionTokenHandlerClientset := connectionTokenHandlerClient.NewForConfigOrDie(config)
+	knativeEventingClientSet := eventing.NewForConfigOrDie(config)
+	//httpSourceClientset := sourcesclientv1alpha1.NewForConfigOrDie(config)
```
Comment (crabtree) [inline tests/end-to-end/external-solution-integration/internal/scenario/connectivity_adapter_e2e/connectivity_adapter_e2e.go]: Leftover?

File: tests/end-to-end/external-solution-integration/README.md [hunk 17]
```diff
@@ -42,7 +42,28 @@ The test performs the following actions:
 11. Sends an event to the Application Gateway. 
 12. Verifies if the call from the lambda reached the test service.
 
-### Environment variables
+### Compass connectivity adapter test
+
+This scenario uses Compass and Connectivity Adapter to register an Application and its APIs and Events.
+Connectivity Adapter is a component which translates Application Registry and Application Connector REST API to Compass GraphQL API.
```
Comment (majakurcius) [inline tests/end-to-end/external-solution-integration/README.md]:
```suggestion
Connectivity Adapter is a component which translates the Application Registry and Application Connector REST API to the Compass GraphQL API. 
```

File: tests/end-to-end/external-solution-integration/README.md [hunk 18]
```diff
@@ -42,7 +42,28 @@ The test performs the following actions:
...
+
+### Steps
+
+The test performs the following actions:
+1. Registers an empty Application in Compass.
+2. Creates the ApplicationMapping CR for the created Application in the `connectivity-adapter-e2e` Namespace.
+3. Deploys a lambda function in the `connectivity-adapter-e2e` Namespace.
+4. Starts a test service in the `connectivity-adapter-e2e` Namespace. The lambda function calls it when it receives an event.
+5. Connects the Application through the Connectivity Adapter with the client certificates. 
+6. Registers Service for the Application using Connectivity Adapter
```
Comment (majakurcius) [inline tests/end-to-end/external-solution-integration/README.md]:
```suggestion
6. Registers a service for the Application using Connectivity Adapter.
```

File: tests/end-to-end/external-solution-integration/README.md [hunk 19]
```diff
@@ -42,7 +42,28 @@ The test performs the following actions:
...
+7. Creates ServiceInstances for ServiceClasses registered by Compass Runtime Agent (one for the API and one for the Event services).
+8. Creates a ServiceBinding for the API ServiceInstance.
+9. Creates the ServiceBindingUsage CR of that binding for the deployed lambda function. 
+10. Creates a Subscription for the lambda function, so that it is subscribed to the events exposed by the Application.
+11. Sends an event to the Application Gateway. 
+12. Verifies if the call from the lambda reached the test service.
+
+## Environment variables for connectivity-adapter-e2e and compass e2e
```
Comment (majakurcius) [inline tests/end-to-end/external-solution-integration/README.md]:
```suggestion
## Environment variables for connectivity-adapter-e2e and compass-e2e
```

File: tests/end-to-end/external-solution-integration/README.md [hunk 20]
```diff
@@ -75,10 +96,12 @@ When you run the test, these actions are performed in the order listed:
...
 ### Run against Kyma cluster on Minikube 
-1. Add an entry to your system's `/etc/hosts` that maps the `counter-service.kyma.local` to `127.0.0.1` 
-2. Run the test using the following command:
+1. Add an entry to your system's `/etc/hosts` that maps the `counter-service-{testID}.kyma.local` to `minikube cluster IP`,
+e.g.: for *compass-e2e-test*, it will be counter-service-compass-e2e-test.kyma.local 
+2. Set required ENVs
+3. Run the test using the following command:
     ```
-    go run ./cmd/runner e2e
+    go run ./cmd/runner {name-of-test}
```
Comment (majakurcius) [inline tests/end-to-end/external-solution-integration/README.md]:
```suggestion
    go run ./cmd/runner {test-name}
```

File: tests/end-to-end/external-solution-integration/README.md [hunk 21]
```diff
@@ -75,10 +96,12 @@ When you run the test, these actions are performed in the order listed:
 ## Run the test locally
 
 ### Run against Kyma cluster on Minikube 
-1. Add an entry to your system's `/etc/hosts` that maps the `counter-service.kyma.local` to `127.0.0.1` 
-2. Run the test using the following command:
+1. Add an entry to your system's `/etc/hosts` that maps the `counter-service-{testID}.kyma.local` to `minikube cluster IP`,
+e.g.: for *compass-e2e-test*, it will be counter-service-compass-e2e-test.kyma.local
```
Comment (majakurcius) [inline tests/end-to-end/external-solution-integration/README.md]:
```suggestion
1. Add an entry to your system's `/etc/hosts` that maps the `counter-service-{testID}.kyma.local` to `minikube cluster IP`. For example, for `compass-e2e-test`, map `counter-service-compass-e2e-test.kyma.local`. 
```

**Pull request merged**